Fix #961: Removed old visualization code from evaluate.py

Bhavya1604 opened the pull request:
fixes #961 
fixes #964

Removing root_dir from evaluate_image_boxes and evaluate_boxes.
Eliminating unnecessary visualize.plot_predictions and cv2.imwrite calls.

Co-authored-by: Samia Haque Tisha <samiatisha35@gmail.com>

henrykironde reviewed (changes requested):
Thanks @Bhavya1604 for getting to work on this PR. I expected some changes in the tests too.

Bhavya1604 commented:
@henrykironde Thanks for the review,  I’ll update the tests and push the changes soon.

Bhavya1604 commented:
I have made the following changes:

1. Removed root_dir from evaluate.py.
2. Removed all old visualization code from evaluate.py.
3. Updated the corresponding tests:
    - test_evaluate.py
    - test_main.py
4. Removed root_dir from def evaluate in main.py.

All tests have passed on my local system.

Let me know if any further changes are needed!

henrykironde commented on tests/test_evaluate.py:
```diff
@@ -113,9 +106,8 @@ def test_point_recall_image(root_dir, tmpdir):
     img_path = get_data("OSBS_029.png")
     if root_dir == "tmpdir":
```
Do we still root_dir here?

Bhavya1604 replied:
No, it's not required. I missed it thanks for catching that! I'll update it now.

henrykironde reviewed (commented):
There are just a few things left. An I have commented on one. The other is running yapf to clean up the code. That is why it is failing

henrykironde commented on tests/test_evaluate.py:
```diff
@@ -108,42 +101,6 @@ def test_compute_class_recall(sample_results):
     assert evaluate.compute_class_recall(sample_results).equals(expected_recall)
 
 
-@pytest.mark.parametrize("root_dir", [None, "tmpdir"])
```
@Bhavya1604 why are we removing _point_recall_image_ tests ?

Bhavya1604 replied:
From my understanding the purpose of this test was to verify that visualization worked correctly in both cases with tempdir or without root_dir. Since we are removing visualization components, I believe this test is no longer needed.

Please let me know if I’ve misunderstood anything or if you’d prefer modifying it. I’m sorry if I misunderstood anything.

henrykironde commented on src/deepforest/evaluate.py:
```diff
@@ -222,15 +195,13 @@ def evaluate_boxes(predictions, ground_df, root_dir, iou_threshold=0.4, savedir=
 
 
-def _point_recall_image_(predictions, ground_df, root_dir=None, savedir=None):
+def _point_recall_image_(predictions, ground_df):
```
If we are not testing the function, and we do not need it, why is the function still here ?

Bhavya1604 replied:
I apologize for my confusion. After a closer look at the test_point_recall_image test, I realized I misunderstood its purpose. I will add back the necessary part and remove just the root_dir part

henrykironde commented:
Thanks @Bhavya1604 and @Samia35-2973 for the changes to #965 and #964. We have combined those changes.
